### PR TITLE
TTOOLS-627 Some display improvements, added name validation

### PR DESCRIPTION
**Pull request by mdrillin**
- Added name validation functions for virtualizations and views to WithVirtualizationHelpers
- changes to VirtualizationCreatePage and SelectNamePage to utilize the name validation
- Fix validation behavior of AutoForm usage in a few places 
- Some rearranging of VirtualizationsPage and VirtualizationViewsPage to improve display while loading.
- minor changes to icons to make more consistent

### CODE REVIEW

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/1987/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

File: app/ui-react/syndesis/src/modules/data/pages/VirtualizationViewsPage.tsx [hunk 1]
```diff
@@ -187,82 +187,88 @@ export class VirtualizationViewsPage extends React.Component<
...
-                                              variant={'light'}
-                                              noPadding={true}
-                                            >
-                                              <VirtualizationNavBar
-                                                virtualization={virtualization}
+                                          <WithLoader
+                                            error={error}
+                                            loading={!hasData}
+                                            loaderChildren={
+                                              <ViewListSkeleton
```
Comment (gashcrumb) [inline app/ui-react/syndesis/src/modules/data/pages/VirtualizationViewsPage.tsx]: Did you figure out that black bar?  Here I think one option is you could put a `PageSection` around this *or* maybe better use the `PageLoader` [component](https://github.com/syndesisio/syndesis-react/blob/master/app/ui-react/packages/ui/src/Layout/PageLoader.tsx) instead that @riccardo-forina  had mentioned.  The PageLoader shows a spinner though.  Other option could be to borrow [this guy](https://github.com/syndesisio/syndesis-react/blob/master/app/ui-react/packages/ui/src/Integration/IntegrationsListSkeleton.tsx) it's used on the main integration list page and we also borrowed it for the oauth settings page.

File: app/ui-react/syndesis/src/modules/data/pages/ViewEditPage.tsx [hunk 2]
```diff
@@ -172,7 +172,7 @@ export class ViewEditPage extends React.Component<{}, IViewEditPageState> {
                           <WithLoader
                             error={error}
                             loading={!hasData}
-                            loaderChildren={<Loader />}
+                            loaderChildren={<PageLoader />}
```
Comment (gashcrumb) [inline app/ui-react/syndesis/src/modules/data/pages/ViewEditPage.tsx]: Oh this was the one, wasn't it.  Ignore my other comment then :-)

File: app/ui-react/syndesis/src/modules/data/pages/VirtualizationViewsPage.tsx (hunk 1 above)
Comment (mdrillin) [reply app/ui-react/syndesis/src/modules/data/pages/VirtualizationViewsPage.tsx]: Yeah, this is one of the pages with that issue yet.  I'll look at fixing in another PR.  Thx!

**Pull request merged**